fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/OCFLPersistenceConfig.java

```diff
@@ -42,4 +48,25 @@ public MutableOcflRepository repository() {
         final OCFLConstants constants = new OCFLConstants();
         return createRepository(constants.getStorageRootDir(), constants.getWorkDir());
     }
+
+    @Bean
+    public OCFLConstants ocflConstants(){
+        return new OCFLConstants();
+    }
+
+    @Bean
+    public DataSource dataSource(final OCFLConstants ocflConstants) {
+        final var dataSource = new DriverManagerDataSource();
+        final var workDir = ocflConstants.getWorkDir().getAbsolutePath();
+        dataSource.setUrl("jdbc:h2:" + workDir + File.separator + "containment.idx;FILE_LOCK=SOCKET");
```

Comment:
How do we configure Fedora to use MySQL, for example?

Reply:
The H2 in memory index was meant to exist to run tests and so you could run a Fedora with a very low bar. This is elevating it and I'm not sure we want to do that.

Reply:
The purpose of this PR is to solve the immediate problem not being able to restart fedora without having to blow away the work directory.  I think there are two other tasks which should be addressed separately (which I will create jiras for):  
1. move the FedoraToOCFLIndex state into the database (currently map is written to disk)
2. make the datasource configurable (add mysql, postgresql, etc)

Reply:
https://jira.lyrasis.org/browse/FCREPO-3308
11:22
https://jira.lyrasis.org/browse/FCREPO-3309